### DISCUSSION TITLE
Are there any best practices or coding standards you recommend for new contributors?

### DISCUSSION THREAD

**Discussion by Codingnaveen46** [sanjay-kv/Open-source-Practice#1510] [Q&A] [answered]
Are there any best practices or coding standards you recommend for new contributors?

**Comment by codeftsaikat** [comment]
When new contributors join a project, it's crucial to provide them with clear coding standards and best practices to ensure consistency and maintainability. For the Devlabs repository, or any React-based project, here are some key recommendations:
"Follow the Project's Code Style"
"Write Descriptive Commit Messages"
"Organize Imports"
"Consistent Naming Conventions"
"Avoid Duplicate Code"

**Comment by imsudiptaa** [comment, marked as answer]
Yes, adopting best practices and following coding standards can make your contributions more effective and maintainable, especially as a new contributor. Here are some highlighted points to note :

1. Understand the project : Read the documentation, contribution guide, spend time exploring the file structure, naming conventions, and coding patterns. 
2. Code Style and Formatting : Use the formatting and coding style the project follows
3. Write Clean and Readable Code :  Add comments to explain the why, not the what, of complex code sections, use meaningful commits.
4. Git Practices : Always work on a separate branch for each feature or bug fix, Keep your branch updated with the main branch to avoid conflicts, Provide a clear description of the changes, Reference any related issue numbers, Include screenshots or logs for UI changes or bug fixes.
5. Seek Feedback and Learn : Don't hesitate to ask for clarification if you are unsure about something. Take code reviews constructively and use them to improve your skills.


Hope it helps.
Happy Coding !!

**Reply by Codingnaveen46** [reply]
Thank you for sharing these valuable insights on contributing effectively as a new contributor. I completely agree that understanding the project and adhering to coding standards is critical for maintainable and impactful contributions.
